File: C_Programming/Practise_Programs/PP_154.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_154
//
//  Description       : This program checks whether a bit at a given
//                      position in an unsigned integer is ON or OFF
//                      using a bit mask.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 18/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h> 

typedef unsigned int UINT;

int main()
{
    UINT iNo = 0;
    UINT iMask = 0x1;
    UINT iPos = 0;
    UINT iAns = 0;

    printf("Enter Number : \n");
    scanf("%d", &iNo);

    printf("Enter Position : \n");
    scanf("%d", &iPos);    

    iMask = iMask << (iPos - 1);

    iAns = iNo & iMask;

    if(iAns == iMask)
    {
        printf("Bit is ON\n");
    }
    else
    {
        printf("Bit is OFF\n");
    }

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      12
//      3
//
//  Sample Output :
//
//      Enter Number :
//      Enter Position :
//      Bit is ON
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_154.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_154
//
//  Description       : This program accepts a number from the
//                      user and displays each number followed
//                      by a star using a loop.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 03/09/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_154
{
    public static void Display(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
            System.out.print(iCnt + "\t" + "*\t");
        }
        System.out.println();
    }
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the elements : 5
//
//  Sample Output :
//
//      1	*	2	*	3	*	4	*	5	*
//
//////////////////////////////////////////////////////////////////